### PR TITLE
[quantization][opencl]QuantizeAndDequantize

### PR DESCRIPTION
**Pull request by ZchiPitt**
Fix this test so that there are actual quantize and dequantize instructions executed, as they were optimized out previously.

### CODE REVIEW

File: tests/unittests/OperatorTest.cpp [hunk 1]
```diff
@@ -928,21 +928,35 @@ TEST_P(InterpAndCPU, ScatterAssignQuantized) {
...
+  Tensor input2(ElemKind::FloatTy, {1, 4});
+  input1.getHandle() = {1, 1.2f, 0.5f, 1.3f};
+  input2.getHandle() = {1.9f, 5.2f, 3.5f, 6.3f};
 
   auto *A = mod_.createVariable(ElemKind::FloatTy, {1, 4}, "A",
                                 VisibilityKind::Public);
+  auto *B = mod_.createVariable(ElemKind::FloatTy, {1, 4}, "B",
+                                VisibilityKind::Public);
 
   auto qType = mod_.uniqueType(ElemKind::Int8QTy, {1, 4}, 0.05, -138);
```
Comment (rdzhabarov) [inline tests/unittests/OperatorTest.cpp]: make sure that values from the fp32 variables can be represented by {s,o} = {0.05, -138}

File: tests/unittests/OperatorTest.cpp [hunk 2]
```diff
@@ -928,21 +928,35 @@ TEST_P(InterpAndCPU, ScatterAssignQuantized) {
...
-  auto *quantize = F_->createQuantize("quantize", A, qType);
-  auto *dequantize = F_->createDequantize("dequantize", quantize);
-  auto *result = F_->createSave("save", dequantize);
+  auto *quantizeA = F_->createQuantize("quantize", A, qType);
+  auto *quantizeB = F_->createQuantize("quantize", B, qType);
+  auto *int8Add = F_->createAdd("int8Add", quantizeA, quantizeB);
+  auto *dequantize = F_->createDequantize("dequantize", int8Add);
+  auto *int8Result = F_->createSave("int8Save", dequantize);
```
Comment (rdzhabarov) [inline tests/unittests/OperatorTest.cpp]: a bit misleading name as the result in fact fp32 here.

File: tests/unittests/OperatorTest.cpp [hunk 3]
```diff
@@ -928,21 +928,35 @@ TEST_P(InterpAndCPU, ScatterAssignQuantized) {
...
+  auto *fpResult = F_->createSave("fpSave", fpAdd);
 
   EE_.compile(CompilationMode::Infer, F_);
-  EE_.run({A}, {&inputs});
+  EE_.run({A, B}, {&input1, &input2});
+
+  Tensor int8Out;
+  Tensor fpOut;
+  
+  int8Out.copyFrom(&int8Result->getVariable()->getPayload());
```
Comment (rdzhabarov) [inline tests/unittests/OperatorTest.cpp]: do not need to copy around, just use the payload from "save" nodes.

File: tests/unittests/OperatorTest.cpp (hunk 1 above)
Comment (ZchiPitt) [reply tests/unittests/OperatorTest.cpp]: changed to the same values as in input2, so i can ensure they can be dequantzied

File: tests/unittests/OperatorTest.cpp (hunk 2 above)
Comment (ZchiPitt) [reply tests/unittests/OperatorTest.cpp]: removed the prefix

File: tests/unittests/OperatorTest.cpp (hunk 3 above)
Comment (ZchiPitt) [reply tests/unittests/OperatorTest.cpp]: done

File: tests/unittests/OperatorTest.cpp (hunk 1 above)
Comment (rdzhabarov) [reply tests/unittests/OperatorTest.cpp]: my point is that {0.05, -138} covers floating point range [0.05*(-128 + 138), 0.05 *(127+138)].
I was just asking to make sure that the values from input1 and input2 are within this range :).
I think it's better to have different values for input1 and input2 though.

File: tests/unittests/OperatorTest.cpp [hunk 4]
```diff
@@ -928,21 +928,30 @@ TEST_P(InterpAndCPU, ScatterAssignQuantized) {
...
 
 TEST_P(Operator, QuantizeAndDequantize) {
-  Tensor inputs(ElemKind::FloatTy, {1, 4});
-  inputs.getHandle() = {1, 1.2f, 0.5f, 1.3f};
+  Tensor input1(ElemKind::FloatTy, {1, 4});
+  Tensor input2(ElemKind::FloatTy, {1, 4});
+  input1.getHandle() = {1, 1.2f, 0.5f, 1.3f};
+  input2.getHandle() = {1, 1.2f, 0.5f, 1.3f};
 
   auto *A = mod_.createVariable(ElemKind::FloatTy, {1, 4}, "A",
```
Comment (rdzhabarov) [inline tests/unittests/OperatorTest.cpp]: super nit: you can remove Tensor input1 and just directly initialize variable A,
A->getPayload().getHandle()={1, 1.2f, 0.5f, 1.3f}.
And if so when you run E_.run just omit additional initialization of tensors.

**Comment by nadavrot** [comment]
cool!

**Pull request merged**